### PR TITLE
Add attribute to type definition to apply JRF, 11g only

### LINKED ISSUE #370: Extension templates not getting applied after WDT upgrade
Definition extension templates are no longer applied after upgrading from WDT 0.20 to 0.21. I am testing using WLS10.3.6 with the following typesdef file:
```json
{
    "copyright": "Copyright (c) 2017, 2018, Oracle and/or its affiliates. All rights reserved.",
    "license": "The Universal Permissive License (UPL), Version 1.0",
    "name": "WLS",
    "description": "WLS type domain definitions",
    "versions": {
        "10.3.6": "WLS_11G",
        "12.1.1": "WLS_11G",
        "12.1.2": "WLS_12CR1",
        "12.1.3": "WLS_12CR1",
        "12.2.1": "WLS_12CR2",
        "19.1":   "WLS_19CR1"
    },
    "definitions": {
        "WLS_11G": {
            "baseTemplate": "@@WL_HOME@@/common/templates/domains/wls.jar",
            "extensionTemplates": ["@@ORACLE_HOME@@/Oracle_BI1/common/templates/applications/oracle.bipublisher_template_11.1.1.jar" ],
            "serverGroupsToTarget": [ ],
            "rcuSchemas": [ ]
        },
        "WLS_12CR1": {
            "baseTemplate": "@@WL_HOME@@/common/templates/wls/wls.jar",
            "extensionTemplates": [ ],
            "serverGroupsToTarget": [ ],
            "rcuSchemas": [ ]
        },
        "WLS_12CR2": {
            "baseTemplate": "Basic WebLogic Server Domain",
            "extensionTemplates": [ ],
            "serverGroupsToTarget": [ ],
            "rcuSchemas": [ ]
        },
        "WLS_19CR1": {
            "baseTemplate": "Basic WebLogic Server Domain",
            "extensionTemplates": [ ],
            "serverGroupsToTarget": [ ],
            "rcuSchemas": [ ]
        }
    }
}
```
EDIT: I see the same issue on WDT 0.22 and 0.24.

### PR DESCRIPTION
**Pull request by rakillen**
Fixed problem with null fields in summary logging also.

Fixes #370.

### CODE REVIEW

File: site/type_def.md [hunk 1]
```diff
@@ -0,0 +1,246 @@
+## Domain Type Definitions
+
+WebLogic Server Deploy Tooling has an extensible domain type system.  The three built-in domain types (`WLS`, `RestrictedJRF`, and `JRF`) are defined in JSON files of the same name in the `WLSDEPLOY_HOME/lib/typedefs` directory.  For example, the `JRF` domain type is defined in the `WLSDEPLOY_HOME/lib/typedefs/JRF.json` file whose contents look like those shown below.
```
Comment (rosemarymarano) [inline site/type_def.md]: whose contents look like those shown below. -> with similar content, as shown below.

File: site/type_def.md [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+                "Oracle Enterprise Manager"
+            ],
+            "serverGroupsToTarget": [ "JRF-MAN-SVR", "WSMPM-MAN-SVR" ],
+            "rcuSchemas": [ "WLS", "MDS", "IAU", "IAU_VIEWER", "IAU_APPEND", "OPSS" ]
+        }
+    }
+}
+```
+
+This file tells the Create Domain Tool which templates to use to create the domain, which server groups to target, and even which RCU schemas to create, all based on the version of WebLogic Server installed.  New domain types can be defined by creating a new JSON file with the same structure in the `WLSDEPLOY_HOME/lib/typedefs` directory.  For example, to define a `SOA` domain type for 12.2.1.3, add the `WLSDEPLOY_HOME/lib/typedefs/SOA.json` file with contents like those shown below.
```
Comment (rosemarymarano) [inline site/type_def.md]: based on the version of WebLogic Server installed. -> based on the installed version of WebLogic Server.

with contents like those shown below. -> with similar content, as shown below.

File: site/type_def.md [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+            "serverGroupsToTarget": [ "MY-MAN-SVR" ],
+            "rcuSchemas": [ ]
+        }
+    }
+}
+```
+
+If there are any server groups in the custom template that should be targeted to managed servers, they should be specified in the `serverGroupsToTarget` attribute, similar to `MY_MAN_SVR` in the example above.
+
+### Targeting in Earlier Weblogic Server Versions
```
Comment (rosemarymarano) [inline site/type_def.md]: Weblogic -> WebLogic

File: site/type_def.md [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+
+Templates in WebLogic Server versions prior to 12.2.1 may require use of the `applyJRF` WLST command to correctly target resources to the correct clusters and servers. The default behavior for WebLogic Deploy Tooling is to invoke `applyJRF` only when the `extensionTemplates` list includes JRF templates.
```
Comment (rosemarymarano) [inline site/type_def.md]: may require use of ->  may require the use of

File: site/type_def.md (hunk 1 above)
Comment (rakillen) [reply site/type_def.md]: Fixed

File: site/type_def.md (hunk 2 above)
Comment (rakillen) [reply site/type_def.md]: Fixed

File: site/type_def.md (hunk 3 above)
Comment (rakillen) [reply site/type_def.md]: Fixed

File: site/type_def.md (hunk 4 above)
Comment (rakillen) [reply site/type_def.md]: Fixed

**Pull request merged**